src/client.rs

```diff
@@ -273,13 +267,17 @@ impl Client {
     ///
     /// This performs authorization and then stores the token internally to be used
     /// on other requests.
-    async fn auth(
+    pub async fn auth(
         &mut self,
         image: &Reference,
         authentication: &RegistryAuth,
         operation: RegistryOperation,
     ) -> anyhow::Result<()> {
         debug!("Authorizing for image: {:?}", image);
+        if self.tokens.contains_key(image, operation) {
```

Comment:
Hmm... I see the reasoning behind this change. I think we designed it the other way so the caller can force a token refresh. It makes the code a little less DRY, but it does give the caller a bit more flexibility.

The other thing we need to address at some point is whether the token has expired (which I noticed in your review). Not required for this PR, but it's worth keeping in mind.

Reply:
Should we remove this commit? or replace with public `TokenCache` for user to do the token expire check？

Reply:
oci-distribution should be the one responsible for checking and refreshing auth tokens. I don't think that should be the caller's responsibility.

That being said, this change actively prevents the user from refreshing the auth token, so I think this is a breaking change that should be reverted. Users should be able to manually refresh tokens by calling `client.auth` at-will.